src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -83,28 +88,37 @@ private MaxAndReachablePoints calculateMaxAndReachablePoints(Set<Exercise> exerc
             }
         }
 
-        return new MaxAndReachablePoints(maxPoints, reachableMaxPoints);
+        if (gradingScale != null) {
+            reachablePresentationPoints = presentationPointsCalculationService.calculateReachablePresentationPoints(gradingScale, reachableMaxPoints);
+            maxPoints += reachablePresentationPoints;
+            reachableMaxPoints += reachablePresentationPoints;
+        }
+
+        return new MaxAndReachablePoints(maxPoints, reachableMaxPoints, reachablePresentationPoints);
     }
 
     /**
      * Prepares all entities required for calculateCourseScoresForStudent and calls it to retrieve the total scores for each specified student in the specified course.
      * If there is a single student id in studentIds, the student id will be filtered in the database as an optimization wherever possible.
      *
-     * @param courseId   the id of the course to calculate the total scores for.
-     * @param studentIds the id of the students whose scores in the course will be calculated.
+     * @param course       the course to calculate the total scores for.
+     * @param gradingScale the grading scale with the presentation configuration to use for calculating the scores.
+     * @param studentIds   the id of the students whose scores in the course will be calculated.
      * @return the max and reachable max points for the given course and the student scores with related plagiarism verdicts for the given student ids.
      */
     @Nullable
-    public Map<Long, BonusSourceResultDTO> calculateCourseScoresForExamBonusSource(long courseId, Collection<Long> studentIds) {
+    public Map<Long, BonusSourceResultDTO> calculateCourseScoresForExamBonusSource(Course course, GradingScale gradingScale, Collection<Long> studentIds) {
+        if (course == null) {
+            return null;
+        }
+        long courseId = course.getId();
```

Comment:
```suggestion
        Long courseId = course.getId();
```
As far as I know Long is used for ids